### ISSUE TITLE
[3.3.5a] Mobs not properly assisting nearby allies on engage

### ISSUE THREAD

**Issue by Treeston** [TrinityCore/TrinityCore#15437]
Branch 3.3.5, revision 61085016dc11, TDB 335.59

Pulling a mob inside an instance only causes adjacent mobs of the same family (humanoid/beast/undead/whatever) to assist. This lets you pull only small parts of a pack on their own.
Expected behavior would be to have all adjacent mobs regardless of family assist on engage while inside an instance.

Repro example:
1. Enter Hellfire Ramparts
2. Advance until you reach a pack of 2+ worgs & 2+ orcs.
3. Hit one of the worgs from range (do not OHKO it)
4a. Only the other worg(s) in the pack will aggro to assist the pulled worg.
4b. The orcs in the pack will completely ignore you as if nothing happened. Expected behavior is to have the orcs also assist the pulled worg.

**Comment by MrSmite** [comment]
> Expected behavior would be to have all adjacent mobs regardless of family assist on engage while inside an instance.

This isn't entirely correct, It should be *all linked mobs*. In your example, yes the Orcs should also aggro when you pull the two Wargs. This indicates bad linking.

If you set it to "all nearby mobs" then you end up with aggro chaining problems (which Trinity had) where you pull one mob and it spreads to every mob in the instance. Also, on live in 335 you could pull packs of mobs without aggro from other packs.

**Comment by Saben65** [comment]
A hackfix is to add the creatures to Creature_formations with a groupAI of 2. You can leave the distance and angle at 0 and so long as the leader doesn't have WPs, it won't affect the spawns.

A similar table like this that ignores WPs (for cases where one mob from another family is patrolling such as in Sethekk Halls) could be beneficial.

**Comment by Rushor** [comment]
> A hackfix is to add the creatures to Creature_formations with a groupAI of 2. You can leave the distance and angle at 0 and so long as the leader doesn't have WPs, it won't affect the spawns.

i don't even think that this is a hackfix, maybe blizz scripts all creatures like that manually

**Comment by Saben65** [comment]
I would consider it a hack fix because there are quite a few cases where it won't work. I believe one case is in mana tombs where you have a bird patrolling around a group of humanoids while the humanoids are stationary.

**Comment by pete318** [comment]
> Repro example:
1. Enter Hellfire Ramparts
2. Advance until you reach a pack of 2+ worgs & 2+ orcs.
3. Hit one of the worgs from range (do not OHKO it)
4a. Only the other worg(s) in the pack will aggro to assist the pulled worg.
4b. The orcs in the pack will completely ignore you as if nothing happened. Expected behavior is to have the orcs also assist the pulled worg.

I think in this case, it's a bit more simple. The worg's should be considered pets and linked to an orc each, the aggro on the worg regardless of distance should trigger the orc (and since they patrol together you're ensured to get both orcs and both worgs).

If the orc's are close enough to the group of orcs, since they are linked you should get that whole group too through the humanoid class like proximity aggro.

This was why you'd time it to ensure you didn't get both on retail.

**Comment by Rushor** [comment]
> I would consider it a hack fix because there are quite a few cases where it won't work. I believe one case is in mana tombs where you have a bird patrolling around a group of humanoids while the humanoids are stationary.

should be fixed with https://github.com/TrinityCore/TrinityCore/pull/18733

**Comment by Killyana** [comment]
The PR was merged, formations must be done using groupAI=3

**Comment by offl** [comment]
> A hackfix is to add the creatures to Creature_formations with a groupAI of 2. You can leave the distance and angle at 0 and so long as the leader doesn't have WPs, it won't affect the spawns.
>> i don't even think that this is a hackfix, maybe blizz scripts all creatures like that manually

Yes, they script them manually. And in Ramparts only formations are required. It's a pure DB issue.

> I would consider it a hack fix because there are quite a few cases where it won't work. I believe one case is in mana tombs where you have a bird patrolling around a group of humanoids while the humanoids are stationary.

I think you mean Sethekk Halls, because there's no birds in Mana Tombs.
I have did formations for Sethekk Halls based on retail behavior and the birds just shouldn't have any formations, only Arrakoa NPCs should have them. On retail the birds can be pulled without aggroing other Arrakoa NPCs. The same for other instances, single-pathing NPCs in most cases shouldn't have formations.
